modules/special-rules/pages/barrage.adoc

```diff
@@ -4,29 +4,62 @@ These weapons can saturate a wide area of the battlefield with intense salvoes o
+When you place Blast markers, {barrage} weapons contribute to your detachment's total Firepower, but not to the number of super-heavy weapons.
 
-+[TODO: It'd be good to rationalise this rule a bit, given that (1) infantry units don't always have five individual models on them, and (2) some people like to put their vehicles on bases, and (3) it isn't clear how much of a vehicle or war engine should count to resolve disagreement when a template barely 'nicks' such a unit.]+
+It still takes only a single Blast marker to suppress a {barrage} weapon though, just like any other super-heavy weapon.
 ====
-. Count the number of units beneath the template.
-The number of units under the template is your unit's Firepower value for this shot.
 
-. Add this Firepower value to the rest of the Firepower of your detachment for the current attack. 
+.Intermingled detachments under a Barrage template
+[NOTE]
+====
+After you place to template to cover as many units in the target detachment as possible, it may be that the template also covers some units from another detachment.
 
-IMPORTANT: For the placement of Blast markers, count a {barrage} shot along with regular Firepower and not as a super-heavy weapon.
+Make a separate count of the number of units of the other detachment that are under the template.
+After you have resolved the attack against the target detachment, use the separate count to resolve the attack against the other detachment, as if it were a separate attack.
+====
 
 .A classic {barrage} unit
 ====
 The Space Marine Whirlwind armoured vehicle is armed with a Whirlwind missile launcher.
 What this vehicle lacks in range it can make up for in speed and the ability to rain a destructive barrage of missiles down onto an area of the battlefield.
-====
\ No newline at end of file
+====
+
+.Adding up {barrage} Firepower
+====
+You shoot three Space Marine Whirlwind units at an Ork Warband detachment that is partially intermingled with a Kult of Speed detachment.
+
+.Adding up {barrage} Firepower. Numbered rectangles are Ork Boyz units; solid circles are Gunwagon units; semi-opaque circles are Barrage templates. Green units are in the Ork Warband detachment; red units are in the Kult of Speed detachment.
+image::barrage-example-1a.png[One Barrage template that covers Ork Infantry and Vehicle units,320]
+
+You place one Barrage template for all three Whirlwinds.
+
+In total, the template covers three units from the target detachment:
+
+* The template covers Ork Boyz (1) and (2), and the Gunwagon.
+* Ork Boyz (3) is not covered by the template because their centre is not under the template.
+* We ignore Ork Boyz (4) as it is not in the target detachment.
+// TODO: But we don't ignore it completely do we? We'll resolve a separate attack on that detachment at 3 Firepower, won't we?
+
+So, each Whirlwind adds 3 Firepower to the total for your detachment.
+As the Whirlwinds have no other help, the total is 9 Firepower against the Ork Warband detachment.
```

Comment:
How about this:
```suggestion
In total, the template covers three units from the target detachment:

* The template covers Ork Boyz (1) and (2), and the Gunwagon.
* Ork Boyz (3) is not covered by the template because their centre is not under the template.
* Ork Boyz (4) is in a different detachment, the Kult of Speed — so we make a separate count to resolve the attack against that detachment.

So, against the target detachment, each Whirlwind adds 3 Firepower to the total for your detachment's attack.
As the Whirlwinds have no other help, the total is 9 Firepower against the Ork Warband detachment.

After you resolve that, you also resolve the attack against the Kult of Speed detachment.
Each Whirlwind contributes 1 Firepower in this case, for a total of 3 Firepower.
```

Reply:
I'm pretty sure this works, so I'll commit it. You can still highlight a problem if you see one.